.github/workflows/step_change-detection.yml

```diff
@@ -0,0 +1,60 @@
+name: Detect file changes
+on:
+  workflow_call:
+    outputs:
+      docs_changed:
+        description: Documentation changed
+        value: ${{ jobs.change-detection.outputs.docs_changed }}
+      base_changed:
+        description: Base Swig files changed
+        value: ${{ jobs.change-detection.outputs.base_changed }}
+      langs:
+        description: Swig languages affected
+        value: ${{ jobs.change-detection.outputs.langs }}
+
+
+jobs:
+  change-detection:
+    name: Gather commit changes
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+      - name: Install fmf
+        run: python3 -m pip install fmf
+      - name: Check documentation
+        id: docs
+        uses: tj-actions/changed-files@v38
+        with:
+          files: |
+            Doc/**
+      - name: Check Swig base
+        id: base
+        uses: tj-actions/changed-files@v38
+        with:
+          files: |
+            Source/**
+            Lib/std/**
+            Lib/typemaps/**
+            Lib/xml/**
+            Lib/*
+      - name: Check Swig language
+        id: lang
+        uses: tj-actions/changed-files@v38
+        with:
+          path: Lib
+          files: |
+            */**
+          files_ignore: |
+            .fmf/*
+            std/**
+            typemaps/**
+            xml/**
```

Comment:
Should also probe changes in per language.
Examples/*
Examples/test-suite/*/**
There is one annoying exception, the schemerunme folder is used by guile.
It was probably planned as a shared folder for scheme based languages.

Reply:
This one is a bit tricky because I was using the path of the library to get the language path. I'll try to convert it to a separate job and get the info from fmf.

Reply:
Ok, found a suitable workflow for that. Only `schemerunme` has this special treatment?

Reply:
> Ok, found a suitable workflow for that. Only schemerunme has this special treatment?

As I check all languages in `Examples/test-suite`, it is the only one at the moment.